src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2D.java

```diff
@@ -804,4 +845,14 @@ public FontRenderContext getFontRenderContext() {
 		calcGfx.addRenderingHints(renderingHints);
 		return calcGfx.getFontRenderContext();
 	}
+
+	// In case a single 2D graphic object has to be written (one or more times)
+	// into a PDF document, this method isn't needed.
+	// If, instead, several 2D graphic objects have to be written (one or more times)
+	// into a PDF document, the graphic state has to be saved after a graphic object
+	// has been written for the last time and a new graphic object has to be created
+	public void saveGraphicsState() throws IOException
```

Comment:
That seems very strange. It would be awesome if you could create a small testcase / testdriver for that. Just put it in src/test/java/de/rototor/pdfbox/graphics2d. And feel free to add dependencies to JFreeChart or whatever you need for that to the pom.xml with test-scope.

And this shouldn't even work, as on the contentStream close() has already been called after dispose()... You might writing directly into the main content stream of the document... I've got to investigate what PDFBox does here.

I'll try to get to this later today when I'm at home.

Also a side note: You know that you can use https://github.com/danfickle/openhtmltopdf to create your reports, so you don't have to mess around with PDFBox directly? I've integrated this library there, so that you can put your charts as HTML-<object> tags into the report. A sample how to integrate your own renderer are here: https://github.com/danfickle/openhtmltopdf/blob/open-dev-v1/openhtmltopdf-examples/src/main/java/com/openhtmltopdf/testcases/TestcaseRunner.java - If it's not clear how this works feel free to ask.